### DISCUSSION TITLE
How can i pass a python object to Rust, and invoke a callback function

### DISCUSSION THREAD

**Discussion by unigrav-tech** [PyO3/pyo3#2390] [Questions]
If I have a python class A 
class A:
   def cb(**kwargs):
       dict = {...}
        return dict

And I want to pass an instance of class A to Rust, and invoke the  cb() function in the Rust, like:

let mut args = Hashmap::new()
...
let result = instanceOfA.cb(args)
...

**Comment by adamreichold** [comment]
Your Rust function can accept `instanceOfA: &PyAny` and invoke `cb` via

```rust
let res = instanceOfA.call1(args)?;
```
